pyccel/parser/semantic.py

```diff
@@ -2362,9 +2362,12 @@ def _visit_Return(self, expr, **settings):
             if not (isinstance(r, PyccelSymbol) and r == v):
                 a = Assign(v, r)
                 a.set_fst(expr.fst)
+                a = self._visit_Assign(a)
+                ind = a.rhs.get_attribute_nodes(IndexedElement)
+                if len(ind) == 0 and not isinstance(a.rhs, IndexedElement):
+                    a.lhs.is_temp = True
```

Comment:
Im not sure about this . if the rhs of the Assign is not an indexed element and it don't contain an indexed element
you take it as temporary variable .
and in the printing stage you search for deallocate node 
I think this will not work for arguments example 
```PYTHON
@types('int[:]')
def func(c):
    return c[0]
```